tests/testthat/test-backend-mssql.R

```diff
@@ -200,3 +89,52 @@ test_that("custom lubridate functions translated correctly", {
   expect_equal(trans(quarter(x, with_year = TRUE)), sql("(DATENAME(YEAR, `x`) + '.' + DATENAME(QUARTER, `x`))"))
   expect_error(trans(quarter(x, fiscal_start = 5)))
 })
+
+with_options(dbplyr_table_num = 0, verify_output("sql/mssql.txt", {
+  mf <- lazy_frame(x = 1:3, con = simulate_mssql())
+
+  "# filter and mutate translate is.na correctly"
+  mf %>% head()
+
+  mf %>% mutate(z = is.na(x))
+
+  mf %>% mutate(z = !is.na(x))
+
+  mf %>% filter(is.na(x))
+
+  "Invalid, can we reliably detect that the return value is logical?"
```

Comment:
These should stay as separate tests; no need to cram unrelated tests into a single `verify_output()`.